pyqtorch/primitive.py

```diff
@@ -258,23 +253,25 @@ def __init__(self, control: int | tuple[int, ...], target: tuple[int, ...]):
         super().__init__(OPERATIONS_DICT["CSWAP"], target)
         self.control = (control,) if isinstance(control, int) else control
         self.target = target
-        self.qubit_support = self.control + self.target
+        self.orig_support = self.control + self.target
+        self.qubit_support = tuple(sorted(self.orig_support))
 
     def extra_repr(self) -> str:
         return f"control:{self.control}, target:{self.target}"
 
 
 class ControlledOperationGate(Primitive):
     def __init__(self, gate: str, control: int | tuple[int, ...], target: int):
-        self.control = (control,) if isinstance(control, int) else control
+        self.control: tuple = (control,) if isinstance(control, int) else control
         mat = OPERATIONS_DICT[gate]
         mat = _controlled(
             unitary=mat.unsqueeze(2),
             batch_size=1,
             n_control_qubits=len(self.control),
         ).squeeze(2)
         super().__init__(mat, target)
-        self.qubit_support = self.control + (self.target,)  # type: ignore[operator]
+        self.orig_support = self.control + (self.target,)  # type: ignore [operator]
```

Comment:
Yes but it goes into a full sequence of mypy complaining in other places, and fully fixing it implies changing a few more things of `Primitive`. It will be handled with #240.